content/tutorials/channel-rewind.textile

```diff
@@ -0,0 +1,217 @@
+        }
+    });
+}
+```
+
+If you run this file in your browser now, you should see the realtime data coming through from the weather data stream. Well, even if a new update wasn't published just when you run this app, you can still see the most recent update previously sent on this channel, thanks to the channel's rewind param.
+
+h3(#using-sse). Option 2: Consume rewound and new data using Ably's SSE protocol adapter
+
+Let us now look at consuming the same data stream with the rewind parameter, but only this time using Server-Sent Events (SSE) instead of Ably's native client SDKs. Note that since we'll use the SSE protocol, we don't need to use any client SDK at all. We'll make use of the @EvenSource()@ API that comes with all major browsers by default. Replace the script in the previous step with the following:
+
+```[html]
+<script>
+    var $resultWeather = $('#result-open-weather'),
+            oldTemperature = -50,
+            channelWeather,
+            eventSource;
+    var apiKey = "<YOUR-API-KEY>"
+    $('select#city-id').on('change', function () {
+            eventSource.close();
+            subscribeOpenWeather($('select#city-id').val());
+        });
+    subscribeOpenWeather($('select#city-id').val());
+    function subscribeOpenWeather(id){
+      var channelWeather = "[product:ably-openweathermap/weather?rewind=1]weather:" + id
+      var URL = `https://realtime.ably.io/sse?v=1.1&key=${apiKey}&channels=${channelWeather}`
+      eventSource = new EventSource(URL);
+      eventSource.onmessage = function(msg) {
+          var message = JSON.parse(msg.data);
+          var weatherData = JSON.parse(message.data);
+          $resultWeather.text((weatherData.main.temp - 273.15).toFixed(2) + '°C with ' + weatherData.weather[0].description);
+      }
+      eventSource.onerror = function(error) {
+          if(error.data) {
+              $resultWeather.text(`Error: ${error.data}`);
+          }
+      }
+    }
+</script>
+```
+
+If you run this file in your browser now, you should see the realtime data coming through from the weather data stream via SSE. Notice that we haven't used any client SDKs here. Also, we get the latest message that was previously published on that channel, even if a new update isn't available at the time.
+
+h3(#using-mqtt). Option 3: Consume rewound and new data using Ably's MQTT protocol adapter
+
+You can use any MQTT library to publish and receive messages to and from Ably. For this tutorial, we'll make use of the "MQTT NPM library":https://www.npmjs.com/package/mqtt, but feel free to use any that you like.
+
+Create a new javascript file, say @main.js@. We'll begin by requiring the MQTT library and setting the correct options for our client to use Ably:
+
+```[javascript]
+var mqtt = require('mqtt');
+var options = {
+    keepalive: 30,
+    username: '<FIRST-HALF-OF-YOUR-ABLY-API-KEY>',
```

Comment:
Can't remember the official names for first half and second half, but whatever they are we should use those + maybe a link which explains how a key is broken up? Maybe also a link somewhere around here to the Ably MQTT docs